component/service/src/main/java/org/exoplatform/social/rest/impl/attachment/AttachmentRest.java

```diff
@@ -250,6 +243,51 @@ public Response getAttachment(
     }
   }
 
+  @DELETE
+  @Path("{objectType}/{objectId}/{fileId}")
+  @RolesAllowed("users")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Delete a file linked to a given object identified by its id", description = "Delete a file linked to a given object identified by its id", method = "DELETE")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input"),
+      @ApiResponse(responseCode = "401", description = "Unauthorized request"),
+      @ApiResponse(responseCode = "404", description = "Not found"), })
+  public Response deleteAttachment(
+                                   @Parameter(description = "Object type : activity, task, note...", required = true)
+                                   @PathParam("objectType")
+                                   String objectType,
+                                   @Parameter(description = "Identifier of object from which file will be deleted", required = true)
+                                   @PathParam("objectId")
+                                   String objectId,
+                                   @Parameter(description = "Identifier of file will be deleted", required = true)
+                                   @PathParam("fileId")
+                                   String fileId) {
+
+    if (StringUtils.isBlank(objectType)) {
+      return Response.status(Status.BAD_REQUEST).entity("Object type is required").build();
+    }
+    if (StringUtils.isBlank(objectId)) {
+      return Response.status(Status.BAD_REQUEST).entity("Object Id is required").build();
+    }
+    if (StringUtils.isBlank(fileId)) {
+      return Response.status(Status.BAD_REQUEST).entity("File id is required").build();
+    }
+
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId();
+    if (currentIdentityId <= 0) {
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
```

Comment:
Impossible case, you already added the annotation `@RolesAllowed`
```suggestion
```

Reply:
done!